common/imageloading/src/commonMain/kotlin/app/tivi/common/imageloading/ImageLoader.kt

```diff
@@ -0,0 +1,67 @@
+// Copyright 2023, Christopher Banes and the Tivi project contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package app.tivi.common.imageloading
+
+import app.tivi.app.ApplicationInfo
+import app.tivi.util.Logger
+import coil3.ImageLoader
+import coil3.PlatformContext
+import coil3.annotation.ExperimentalCoilApi
+import coil3.disk.DiskCache
+import coil3.fetch.NetworkFetcher
+import coil3.intercept.Interceptor
+import coil3.memory.MemoryCache
+import coil3.util.Logger.Level
+import okio.Path.Companion.toPath
+
+@OptIn(ExperimentalCoilApi::class)
+internal fun newImageLoader(
+  context: PlatformContext,
+  interceptors: Set<Interceptor>,
+  applicationInfo: ApplicationInfo,
+  logger: Logger,
+  debug: Boolean = false,
+): ImageLoader {
+  return ImageLoader.Builder(context)
+    .components {
+      add(NetworkFetcher.Factory())
+      interceptors.forEach(::add)
+    }
+    .memoryCache {
+      MemoryCache.Builder()
+        .maxSizePercent(context, percent = 0.25)
+        .build()
+    }
+    .diskCache {
+      DiskCache.Builder()
```

Comment:
If you plan to run this on the web you should have this return a null `DiskCache` on JS like [here](https://github.com/coil-kt/coil/blob/main/samples/shared/src/commonMain/kotlin/sample/common/imageLoader.common.kt#L41).

It's not possible to write to the file system on the web so trying to write to it will (currently) throw.

Reply:
No JS target as yet (and I doubt I'll ever bother).